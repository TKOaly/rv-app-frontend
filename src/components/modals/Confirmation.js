import './styles/Confirmation.scss';
import { closeModal } from '../../reducers/modalReducer';
import { connect } from 'react-redux';
import { increaseBalance } from '../../reducers/userReducer';
import { resetAmount, toggleConfirmationVisibility } from '../../reducers/depositReducer';
import React from 'react';
import moneyFormatter from '../../services/moneyFormatter';

export class Confirmation extends React.Component {
    componentDidMount = () => {
        this.oldFocusedElement = document.activeElement;
        this.dialog.focus();
    };

    handleConfirm = (event) => {
        event.preventDefault();
        this.props.increaseBalance(this.props.token, this.props.depositAmount);
        this.props.closeModal();
    };

    handleCancel = (event) => {
        event.preventDefault();
        this.props.toggleConfirmationVisibility(false);
        if (this.oldFocusedElement) {
            this.oldFocusedElement.focus();
        }
    };

    handleKeyDown = (event) => {
        if (event.key === 'Enter') {
            event.preventDefault();
            this.handleConfirm(event);
        } else if (event.key === 'Escape') {
            event.preventDefault();
            this.handleCancel(event);
        }
    };

    dialogRef = (dialog) => {
        this.dialog = dialog;
    };

    render = () => {
        const { depositAmount } = this.props;
        return (
            <div className="confirm-overlay">
                <div className="confirm" tabIndex="0" ref={this.dialogRef} onKeyDown={this.handleKeyDown}>
                    <span className="confirm-title-text">Confirm <b>{moneyFormatter.centsToString(depositAmount)} â‚¬</b> deposit</span>
                    <br />
                    <br />
                    <button className="btn number cancel cancelBtn confirmation-cancel-btn" onClick={this.handleCancel}>
                        Cancel (ESC)
                    </button>
                    <button className="btn number success confirmBtn confirmation-ok-btn" onClick={this.handleConfirm}>
                        OK (ENTER)
                    </button>
                </div>
            </div>
        );
    };
}

const mapStateToProps = (state) => {
    return {
        token: state.authentication.access_token
    };
};

const mapDispatchToProps = {
    increaseBalance,
    resetAmount,
    closeModal,
    toggleConfirmationVisibility
};

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(Confirmation);
