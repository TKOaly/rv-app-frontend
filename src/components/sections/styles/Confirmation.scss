@import "vars";

.confirm-overlay {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background: rgba(0, 0, 0, 0.6);
}

.confirm {
  position: absolute;
  padding-top: 20px;
  padding-bottom: 20px;
  box-sizing: border-box;
  -webkit-box-sizing: border-box;
  width: 70%;
  z-index: 20;
  color: black;
  text-align: center;
  font-size: 20pt;
  left: calc((100% - 70%) / 2);
  top: calc((100% - 70%) / 2);
  background-color: white;
  box-shadow: $box-shadow
}
