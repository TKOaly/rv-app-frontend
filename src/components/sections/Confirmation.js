import './styles/Confirmation.css';
import React from 'react';
import moneyFormatter from '../../services/moneyFormatter';

class Confirmation extends React.Component {
    handleConfirm = (event) => {
        this.props.increaseBalance(this.props.token, this.props.depositAmount);
        this.props.resetAmount();
        this.props.toggleConfirmationVisibility(false);
        this.props.closeModal();
    };

    handleCancel = (event) => {
        this.props.toggleConfirmationVisibility(false);
    };

    render = () => {
        return (
            <div className="confirm-overlay">
                <div className="confirm">
                    Confirm <b>{moneyFormatter.centsToString(this.props.depositAmount)} â‚¬</b> deposit
                    <br />
                    <br />
                    <button className="btn number cancel cancelBtn" onClick={this.handleCancel}>
                        Cancel
                    </button>
                    <button className="btn number success confirmBtn" onClick={this.handleConfirm}>
                        OK
                    </button>
                </div>
            </div>
        );
    };
}

export default Confirmation;
