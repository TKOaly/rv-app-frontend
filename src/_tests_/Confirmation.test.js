import Confirmation from '../components/sections/Confirmation';
// import Enzyme from 'enzyme';
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Confirmation />, div);
    const cmpnt = renderer
        .create(<Confirmation />, {
            createNodeMock: () => {
                return {
                    focus: () => {}
                };
            }
        })
        .toJSON();
    expect(cmpnt).toMatchSnapshot();
});
