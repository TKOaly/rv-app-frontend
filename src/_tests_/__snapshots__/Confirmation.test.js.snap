// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="confirm-overlay"
>
  <div
    className="confirm"
    onKeyDown={[Function]}
    tabIndex="0"
  >
    Confirm 
    <b>
      NaN.NaN
       â‚¬
    </b>
     deposit
    <br />
    <br />
    <button
      className="btn number cancel cancelBtn"
      onClick={[Function]}
    >
      Cancel (ESC)
    </button>
    <button
      className="btn number success confirmBtn"
      onClick={[Function]}
    >
      OK (ENTER)
    </button>
  </div>
</div>
`;
